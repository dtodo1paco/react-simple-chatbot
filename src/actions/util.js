export function is_link(text) {
    if (!/^(f|ht)tps?:\/\//i.test(text)) {
        return false;
    }
    return true;
}
export function add_http(url) {
    if (!is_link()) {
        url = "http://" + url;
    }
    return url;
}