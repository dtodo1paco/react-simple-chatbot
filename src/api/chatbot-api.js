
import { getChatbotClient } from 'config/api-config.js';
async function doSendMessage (message, token) {
    let ret = null;
    try {
        if (token !== null) {
            const client = getChatbotClient(token);
            const response = await client.textRequest(message);
            ret = response.result.fulfillment;
        } else throw new Error("invalid token");
    } catch (err) {
        console.log("ERROR on chatbot call",err);
    }
    return ret;
}

module.exports = { "doSendMessage": doSendMessage };