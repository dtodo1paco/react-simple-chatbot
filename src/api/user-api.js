import axios from 'axios';
import { decrypt } from 'actions/crypt-util.js';

import { API_ROOT } from 'config/api-config';

async function doLogin (user) {
    let ret = null;
    const response = await axios.post(API_ROOT + "/auth/login", user)
        .then((result) => {
            if (result.status === 200 && result.data.auth) {
                ret = result.data;
            } else {
                //console.log("Error response: " + JSON.stringify(result));
            }
        }).catch(result => {
            //console.log("Network error: " + JSON.stringify(result));
        });
    return ret;
}
async function doLogout () {
    let ret = null;
    const response = await axios.get(API_ROOT + "/auth/logout")
        .then((result) => {
            if (result.status === 200 && result.data.auth === false) {
                ret = result.data;
            } else {
                console.log("Error response: " + JSON.stringify(result));
            }
        }).catch(result => {
            //console.log("Network error: " + JSON.stringify(result));
        });
    return ret;
}

async function doAuth (token) {
    let ret = null;
    if (token) {
        axios.defaults.headers.common['x-access-token'] = token;
    }
    const response = await axios.get(API_ROOT + "/auth/me").then((result) => {
        if (result.status === 200 && result.data.password != null) {
            ret = decrypt(result.data.password);
        } else {
            console.log("ERROR: doing auth");
        }
    }).catch(result => {
        console.log("Error on network call", JSON.stringify(result));
    });
    return ret;
}

async function doRegister (user) {
    let ret = null;
    const response = await axios.post(API_ROOT + "/auth/register", user).then((result) => {
        if (result.status === 200 && result.data.auth) {
            ret = result.data;
        } else {
            console.log("ERROR: doing register");
        }
    }).catch(result => {
        console.log("Error on network call", JSON.stringify(result));
    });
    return ret;
}



module.exports = {
    "doLogin": doLogin,
    "doAuth": doAuth,
    "doRegister": doRegister,
    "doLogout": doLogout
};