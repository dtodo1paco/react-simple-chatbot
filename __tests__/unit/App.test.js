jest.autoMockOff();

import React from 'react';
import ReactDOM from 'react-dom';
import TestUtils from 'react-dom/test-utils';

import { getRenderedContent, getComposite} from '../util/render.js';
import { generateUserLogin, generateUserRegister, generateUserLoggedIn } from '../util/fakeGenerator';

import * as ChatbotAPI from 'api/chatbot-api.js';
import * as UserAPI from 'api/user-api.js';
import {
    doAuth as mock_doAuth,
    doLogin as mock_doLogin,
    doRegister as mock_doRegister
    } from '../__mocks__/user-api.js'
import {
    doSendMessage as mock_doSendMessage
    } from '../__mocks__/chatbot-api.js'

const App = require('App').default;

describe('App testing...', () => {

    let exampleUserLogin    = generateUserLogin();
    let exampleUserRegister = generateUserRegister();
    let exampleUserLoggedIn = generateUserLoggedIn();
    beforeEach(function() {
    });

    describe('render', () => {
        it('snapshot', () => {
            const wrapper = shallow(<App userAPI={UserAPI} chatbotAPI={ChatbotAPI} />);
            expect(wrapper).toMatchSnapshot();
        });
        it('mount renders login form by default', () => {
            const component = mount(<App userAPI={UserAPI} chatbotAPI={ChatbotAPI} />);
            const fields = component.find('.field');
            expect(fields.length).toEqual(2);
            const button = component.find('button').first();
            expect(button.text()).toEqual("Login");
        });
        it('mount renders chatbot widget closed if a user is logged in', () => {
            const component = mount(<App userAPI={UserAPI} chatbotAPI={ChatbotAPI} user={exampleUserLoggedIn} />);
            const widget = component.find('.rcw-widget-container');
            expect(widget.length).toEqual(1);
            const opened = component.find('.rcw-opened');
            expect(opened.length).toEqual(0);
        });
    });

    describe('behavoir', () => {
        describe('when loading', () => {
            it('when loading is true, loading icon is shown', () => {
                const wrapper = shallow(<App />);
                const instance = wrapper.instance();
                expect(wrapper.state('thinking')).toBeFalsy();
                expect(wrapper.find('.loading-icon')).toHaveLength(0);
                wrapper.setState({thinking: true});
                expect(wrapper.state('thinking')).toBeTruthy();
                expect(wrapper.find('.loading-icon').length).toEqual(1);
            });
        });
        describe('when login', () => {
            UserAPI.doLogin = mock_doLogin;
            it('when the user login fails, the user is NOT set in state', async () => {
                const wrapper = shallow(<App userAPI={UserAPI} chatbotAPI={ChatbotAPI} />);
                const instance = wrapper.instance();
                expect(wrapper.state('user')).toBe(null);
                let data = await instance.loginHandler(exampleUserLogin);
                wrapper.update();
                expect(wrapper.state('user')).toBeNull();
                expect(wrapper.state().errors).not.toBe(null);
            });
            it('when the user login succeeds, the user is set in state', async () => {
                const wrapper = shallow(<App userAPI={UserAPI} chatbotAPI={ChatbotAPI} />);
                const instance = wrapper.instance();
                expect(wrapper.state('user')).toBe(null);
                let data = await instance.loginHandler(exampleUserLogin);
                wrapper.update();
                expect(wrapper.state().user.username).toBe(exampleUserLogin.username);
                expect(wrapper.state().user.token).not.toBeNull();
                expect(wrapper.state().errors).toBe(null);
            });
        });
        describe('when signup', () => {
            UserAPI.doRegister = mock_doRegister;
            it('when the user register fails, the user is NOT set in state', async () => {
                const wrapper = shallow(<App userAPI={UserAPI} chatbotAPI={ChatbotAPI} />);
                const instance = wrapper.instance();
                expect(wrapper.state('user')).toBe(null);
                let data = await instance.signupHandler(exampleUserRegister);
                wrapper.update();
                expect(wrapper.state().user).toBe(null);
                expect(wrapper.state().errors).not.toBe(null);
            });
            it('when the user register succeeds, the user is set in state', async () => {
                const wrapper = shallow(<App userAPI={UserAPI} chatbotAPI={ChatbotAPI} />);
                const instance = wrapper.instance();
                expect(wrapper.state('user')).toBe(null);
                let data = await instance.signupHandler(exampleUserRegister);
                wrapper.update();
                expect(wrapper.state().user.username).toBe(exampleUserRegister.username);
                expect(wrapper.state().user.nickname).toBe(exampleUserRegister.nickname);
                expect(wrapper.state().user.token).not.toBeNull();
                expect(wrapper.state().errors).toBe(null);
            });
        });

    });
});


//wrapper.find('.rcw-new-message').simulate('input', {target: {value: message}});
//wrapper.find('.rcw-new-message').getElement().value = message;
//wrapper.find('.rcw-new-message').value = message;
//wrapper.find('.rcw-new-message').simulate('keydown', {keyCode: 13, target: {value:message}});
//                wrapper.find('.rcw-new-message').getDOMNode().value = message;
//                wrapper.find('.rcw-new-message').simulate('change');

//wrapper.find('.rcw-new-message').simulate('change',{target: {value: message}});
//let inputs = TestUtils.scryRenderedDOMComponentsWithTag(wrapper, '.rcw-new-message');
//TestUtils.Simulate.change(wrapper.find('.rcw-new-message'), {target: {value: message}});
//wrapper.find('.rcw-new-message').props().onChange({target: {value: message}});
