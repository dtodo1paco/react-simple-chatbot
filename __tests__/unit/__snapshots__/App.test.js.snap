// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App testing... render snapshot 1`] = `
<div
  id="theApp"
>
  <h1>
    Chatbot veterinario
  </h1>
  <User
    loading={false}
    loginHandler={[Function]}
    logoutHandler={[Function]}
    signupHandler={[Function]}
    user={null}
  />
</div>
`;
